lib/s3-client.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+var Promise = require('bluebird');
+var _ = require('underscore');
+var Stream = require('stream');
+var AWS = require('aws-sdk');
+var parsers = require('juttle/lib/adapters/parsers');
+var serializers = require('juttle/lib/adapters/serializers');
+
+var DEFAULT_CONFIG = {
+    format: 'json'
+};
+
+class S3Client {
+    static init(config) {
+        _.extend(DEFAULT_CONFIG, config);
+    }
+
+    constructor(options) {
+        var accessKeyId = options.accessKeyId || DEFAULT_CONFIG.accessKeyId;
+        var secretAccessKey = options.secretAccessKey || DEFAULT_CONFIG.secretAccessKey;
+        var region = options.region || DEFAULT_CONFIG.region;
+        this.format = options.format || DEFAULT_CONFIG.format;
+
+        this.set_or_throw('bucket', options.bucket);
+        this.set_or_throw('key', options.key);
+
+        this.client = Promise.promisifyAll(new AWS.S3({
+            accessKeyId: accessKeyId,
+            secretAccessKey: secretAccessKey,
+            region: region
+        }));
+    }
+
+    set_or_throw(option, value) {
+        if (!value) {
+            throw new Error('S3Client missing required option ' + option);
+        }
+
+        this[option] = value;
+    }
+
+    read(limit) {
```

Comment:
This is `s3-client.js`. The from/to/limit/state arguments are handled in `read.js` at https://github.com/juttle/juttle-s3-adapter/pull/5/files#diff-77960d7f37d725c98368151aa4bc18f7R22.